firstvoices/backend/views/search/custom_search_views.py

```diff
@@ -0,0 +1,45 @@
+from rest_framework import mixins, viewsets
+from rest_framework.response import Response
+
+from backend.views.search.search_query_builder import get_search_query
+from backend.views.search.utils import hydrate_objects
+
+
+class CustomSearchViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
+    http_method_names = ["get"]
+    queryset = ""
+
+    def get_search_params(self):
+        """
+        Function to return search params in a structured format.
+        """
+        input_q = self.request.GET.get("q", "")
+
+        return {"q": input_q}
+
+    def get_raw_objects(self):
+        """
+        Function to build and execute the search query.
+        Returns raw objects returned from elastic-search.
+        """
+        search_params = self.get_search_params()
+        search_query = get_search_query(q=search_params["q"])
+        raw_objects = []
+        response = search_query.execute()
+        if response["hits"]["total"]["value"]:
+            for hit in response["hits"]["hits"]:
+                raw_objects.append(hit)
```

Comment:
Is there are reason not to just return response["hits"]["hits"] ?

Reply:
You're right.  I was doing some manipulation here before, looks like I moved the manipulation but left the loop here. Good catch. Thanks